modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -1194,4 +1234,35 @@ public void createApplication(Application.Name applicationName) {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map<Application.Name, Set<String>> getTagListForApplications(Collection<Application.Name> applicationNames) {
+
+        LOGGER.debug("Retrieving Experiment Tags for applications {}", applicationNames);
+
+        try {
+            List<ListenableFuture<Result<ExperimentTagsByApplication>>> futures = new ArrayList<>();
+
+            for (Application.Name appName : applicationNames) {
```

Comment:
@bargenilesh [cc @AndreaSuckro ] I agree with what you said regarding the IN queries. Also, I agree on the second part too where it might be over-engineering this thing when the API expects low volume and the queries to cassandra are made in parallel using async mode so that network I/O does not bottleneck the overall responsiveness of the API even in case of reasonable number of experiments.

Also, putting this into the cache might make the admin API a little more responsive but I guess we might be able to live with the current responsiveness for now.

@bargenilesh Appreciate your response. Thanks!